addons/stock/models/stock_picking.py

```diff
@@ -300,11 +300,20 @@ class Picking(models.Model):
     product_id = fields.Many2one('product.product', 'Product', related='move_lines.product_id')
     show_operations = fields.Boolean(compute='_compute_show_operations')
     show_lots_text = fields.Boolean(compute='_compute_show_lots_text')
+    has_tracking = fields.Boolean(compute='_compute_has_tracking')
 
     _sql_constraints = [
         ('name_uniq', 'unique(name, company_id)', 'Reference must be unique per company!'),
     ]
 
+    def _compute_has_tracking(self):
+        for picking in self:
+            tracking = picking.mapped('move_line_ids').filtered(lambda m: m.product_id.tracking != 'none')
```

Comment:
i think it'll be enough in most cases to simply run any(m.has_tracking != 'none' for m in picking.move_lines)
can you try it?